### PR TITLE
chore: release dfx 0.23.0

### PR DESCRIPTION
**Pull request by krpeacock**

### CODE REVIEW

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]
Looks like this PR should be to the release-0.23.0 branch

**Comment by ggreif** [comment]
Could `moc` v0.11.3 be snuck into this release? https://dfinity.slack.com/archives/CGA566TPV/p1721510490346709. Just asking sheepishly...

**Comment by dfx-json** [comment]
@ggreif that shouldn't be a problem. @krpeacock could you include moc 0.11.3 in a dfx `0.23.0-beta.1` ?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 # UNRELEASED
 
+# 0.23.0
```
Comment (sesi200) [inline CHANGELOG.md:5]: Is this now correct? There's a 0.22.1 header further below

**Pull request closed without merging**